### PR TITLE
feat: [Consign2] Add basic search

### PR DESCRIPTION
**Pull request by damassi**
Pair with @williardx 

This adds basic unstyled search input functionality to the Consign2 form, and fetches data from diffusion. 

@williardx - Since our pair I moved all of the fetch / business logic outside of the component and into a context (so that the component can remain "dumb") and refactored the state to use the `useReducer` hook. The general rational for using a context here is a) they're cheap to spin up; b) its good to move business logic, fetching, etc., out of components; and c) most importantly, when state concerns cross single component boundaries and spread into the wider tree, as is happening now, one should strive for decoupleability, versus props passing and callbacks; this has great scale and maintainability characteristics through time. 

cc @artsy/grow-devs 

Nothing to really see, but...

search

### CODE REVIEW

File: src/v2/Apps/Consign/Components/GetPriceEstimate/ConsignPriceEstimateContext.tsx [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+              medium
+              mediumHighRangeCents
+              mediumLowRangeCents
+              mediumMidRangeCents
+              midRangeCents
+              sellThroughRate
+              smallHighRangeCents
+              smallLowRangeCents
+              smallMidRangeCents
+              updatedAt
```
Comment (damassi) [inline src/v2/Apps/Consign/Components/GetPriceEstimate/ConsignPriceEstimateContext.tsx:130]: We don't need all of these fields, will clean up in follow-up

File: src/v2/Apps/Consign/Components/GetPriceEstimate/ConsignPriceEstimateContext.tsx [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+            }
+          }
+        `,
+        {
+          artistInternalID,
+          // FIXME: Figure out how to handle this constraint
+          medium: "PAINTING",
```
Comment (damassi) [inline src/v2/Apps/Consign/Components/GetPriceEstimate/ConsignPriceEstimateContext.tsx:137]: @jonallured - we should start thinking about what we're gonna do here

File: src/v2/Apps/Consign/Components/GetPriceEstimate/ConsignArtistAutosuggest.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      theme={{
+        container: {
+          width: "100%",
+        },
+      }}
+    />
+  )
+}
+
+const AutosugggestInput: React.FC = props => {
```
Comment (williardx) [inline src/v2/Apps/Consign/Components/GetPriceEstimate/ConsignArtistAutosuggest.tsx:40]: (non-blocking) Typo here: `Autosuggest` instead of `Autosugggest`

**Comment by williardx** [comment]
LGTM, thanks!

**Pull request merged**